community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py

```diff
@@ -473,27 +473,47 @@ def get_slurm_reservation_maintenance(lkp: util.Lookup) -> Dict[str, datetime]:
         if name != f"{nodes}_maintenance":
           continue
 
+        if nodes not in nodeset_enable_reservation:
+          continue
+
         reservation_map[name] = datetime.fromtimestamp(time_epoch)
 
     return reservation_map
 
 
-def get_upcoming_maintenance(lkp: util.Lookup) -> Dict[str, Tuple[str, datetime]]:
+def get_upcoming_maintenance(lkp: util.Lookup, nodeset_enable_reservation: Set[str]) -> Dict[str, Tuple[str, datetime]]:
     upc_maint_map = {}
 
     for node, properties in lkp.instances().items():
-        if 'upcomingMaintenance' in properties:
+        if 'upcomingMaintenance' in properties and node in nodeset_enable_reservation:
           start_time = datetime.strptime(properties['upcomingMaintenance']['startTimeWindow']['earliest'], '%Y-%m-%dT%H:%M:%S%z')
           upc_maint_map[node + "_maintenance"] = (node, start_time)
 
     return upc_maint_map
 
 
+def get_nodeset_enable_reservation(lkp: util.Lookup) -> Set[str]:
+    nodeset_enable_reservation = set()
+    for nodeset in lkp.cfg.nodeset.values():
+      if nodeset.enable_maintenance_reservation and nodeset.node_count_static:
+        static, _ = lkp.nodenames(nodeset)
```

Comment:
why skip dynamic nodes?

Reply:
For maintenance we should only consider static nodes right ? dynamic nodes when there are re-created there wont be any maintenance issue and once the job is finished they will be deleted.